### PR TITLE
Aggregate endpoints

### PR DESCRIPTION
**Pull request by elee1766**

### CODE REVIEW

File: cl/beacon/validatorapi/post.go [hunk 1]
```diff
@@ -46,14 +52,101 @@ func (v *ValidatorApiHandler) PostEthV1ValidatorSyncCommitteeSubscriptions(w htt
...
+	sub, err := v.Sentinel.SubscribeGossip(r.Context(), subscriptionData)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		l.Info("[beacon api] started aggregating gossip", "subscription", subscriptionData)
+		defer func() {
+			l.Info("[beacon api] aggregation gossip subscriber closed")
+		}()
+		for {
```
Comment (Giulio2002) [inline cl/beacon/validatorapi/post.go:116]: i think that the subnet listening is supposed to expire eventually. also make sure it does not subscribe twice

File: cl/beacon/validatorapi/post.go [hunk 2]
```diff
@@ -46,14 +54,105 @@ func (v *ValidatorApiHandler) PostEthV1ValidatorSyncCommitteeSubscriptions(w htt
...
+	}
+
+	filterString := "/eth2/*/beacon_attestation_*/ssz_snappy"
+	subscriptionData := &sentinel.SubscriptionData{
+		Filter: &filterString,
+		Topics: topics, // TODO: this should be done in advance in the sentinel and recalculated every once in a while
+	}
+	subDuration := time.Duration(v.BeaconChainCfg.SecondsPerSlot*v.NetworkCfg.AttestationPropagationSlotRange*2) * time.Second
+	ctx, cn := context.WithTimeout(r.Context(), subDuration)
+	sub, err := v.Sentinel.SubscribeGossip(ctx, subscriptionData)
```
Comment (Giulio2002) [inline cl/beacon/validatorapi/post.go:111]: plase use sentinelclient

File: cl/beacon/validatorapi/post.go [hunk 3]
```diff
@@ -46,14 +54,105 @@ func (v *ValidatorApiHandler) PostEthV1ValidatorSyncCommitteeSubscriptions(w htt
...
+		After beacon node receives this request,
+		search using discv5 for peers related to this subnet and replace current peers with those ones if necessary.
+		If validator is_aggregator, beacon node must:
+
+		 1. announce subnet topic subscription on gossipsub
+		 2. aggregate attestations received on that subnet
+	*/
+
+	// grab the head state to calculate committee counts with
+	s, err := v.privateGetStateFromStateId("head")
```
Comment (Giulio2002) [inline cl/beacon/validatorapi/post.go:81]: this is wrong. it should be sent to ProcessAttestation instead in the fcu.

File: cl/beacon/validatorapi/post.go [hunk 4]
```diff
@@ -46,14 +54,105 @@ func (v *ValidatorApiHandler) PostEthV1ValidatorSyncCommitteeSubscriptions(w htt
...
+				l.Info("[beacon api] received invalid attestation", "error", err)
+				continue
+			}
+			//
+			err = v.Machine.ProcessAttestations(s, solid.NewDynamicListSSZFromList[*solid.Attestation]([]*solid.Attestation{a}, 1))
+			if err != nil {
+				l.Info("[beacon api] attestation failed validation", "error", err)
+				continue
+			}
+			v.state.AddAttestation(subnetTopicInt, a)
```
Comment (Giulio2002) [inline cl/beacon/validatorapi/post.go:151]: why not add it to the operations pool? we have that.

File: cl/beacon/validatorapi/post.go (hunk 3 above)
Comment (elee1766) [reply cl/beacon/validatorapi/post.go:81]: kk

File: cl/beacon/validatorapi/post.go (hunk 2 above)
Comment (elee1766) [reply cl/beacon/validatorapi/post.go:111]: ok

File: cl/beacon/validatorapi/post.go (hunk 4 above)
Comment (elee1766) [reply cl/beacon/validatorapi/post.go:151]: because i wanted to key them by the topic int as well.

Comment (Giulio2002) [reply cl/beacon/validatorapi/post.go:151]: uhm sure, is it needed later on in the API though? i should check that

**Pull request closed without merging**